### PR TITLE
feat: add UI guidelines to the Frontend readme

### PR DESCRIPTION
**Pull request by Fidesnoella**

### CODE REVIEW

File: frontend/ui-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# UI Guidelines
+At [Tabiya](https://tabiya.tech/), we follow these UI guidelines to ensure consistency and maintainability in our project's user interface.
+
+## Typography
+- Whenever we display text in the UI, we aim to use typography elements. However, we do not specify font, weight, or other font properties to allow for flexibility.
```
Comment (rav3n11) [inline frontend/ui-guidelines.md:5]: But instead we use the Typography variants defined in the application theme, like "h1", "subtitle"...

File: frontend/ui-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Colors
+- We exclusively use colors from the defined theme in the UI. Exceptions are made only if there is a compelling reason to do so.
```
Comment (rav3n11) [inline frontend/ui-guidelines.md:8]: We exclusively use the colors defined in the application theme...

File: frontend/ui-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Layout and Spacing
+- Higher-order components (HOCs) are responsible for managing padding, spacing, and layout of their children components. 
+  Layout is one-directional, from parent to child unless there is a compelling reason for a different approach.
+- HOCs should not manipulate the layout of neighboring components to ensure reusability.
+- When defining margins, rounding, and spacing, we always use the predefined keys from the `tabiyaSpacing` in the theme.
```
Comment (rav3n11) [inline frontend/ui-guidelines.md:14]: When defining spacing options like margins or padding or flex gaps we use the predefined keys from the `tabiyaSpacing` in the application theme. Likewise when defining rounding options like border radius we use the predefined keys from the `tabiyaRounding` in the application theme...

**Review by rav3n11 — CHANGES_REQUESTED** [review]
Good so far. Minor adjustments. We should also add the external style file instead of inline styles once we've had the chance to talk about it...

File: frontend/ui-guidelines.md (hunk 2 above)
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:8]: We only use colors from the defined theme in the UI

Remove: Exceptions are made only if there is a compelling reason to do so.

File: frontend/ui-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+At [Tabiya](https://tabiya.tech/), we follow these UI guidelines to ensure consistency and maintainability in our project's user interface.
+
+## Typography
+- Whenever we display text in the UI, we aim to use typography elements. However, we do not specify font, weight, or other font properties to allow for flexibility.
+
+## Colors
+- We exclusively use colors from the defined theme in the UI. Exceptions are made only if there is a compelling reason to do so.
+
+## Layout and Spacing
+- Higher-order components (HOCs) are responsible for managing padding, spacing, and layout of their children components.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:11]: Higher-Order Components (HOCs) manage padding, spacing, and layout for their child components

File: frontend/ui-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+## Typography
+- Whenever we display text in the UI, we aim to use typography elements. However, we do not specify font, weight, or other font properties to allow for flexibility.
+
+## Colors
+- We exclusively use colors from the defined theme in the UI. Exceptions are made only if there is a compelling reason to do so.
+
+## Layout and Spacing
+- Higher-order components (HOCs) are responsible for managing padding, spacing, and layout of their children components. 
+  Layout is one-directional, from parent to child unless there is a compelling reason for a different approach.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:12]: Control of the layout flows in one direction, from parent to child

File: frontend/ui-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- HOCs should not manipulate the layout of neighboring components to ensure reusability.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:13]: HOCs should avoid altering the layout of neighboring components to promote reusability

File: frontend/ui-guidelines.md (hunk 3 above)
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:14]: When defining margins, rounding, and spacing, we consistently utilize the predefined keys from the tabiyaSpacing theme

File: frontend/ui-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- When defining margins, rounding, and spacing, we always use the predefined keys from the `tabiyaSpacing` in the theme.
+  We avoid using random pixel values (px) unless there is a valid exception.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:15]: We avoid using arbitrary pixel values (px).

File: frontend/ui-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+## Icon Sizes
+- We maintain a consistent use of standard icon sizes specified in the palette throughout the project.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:18]: We use the standard icon sizes specified in the palette to ensure consistency

File: frontend/ui-guidelines.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+## Naming Conventions
+- We maintain naming conventions consistently across the project.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:21]: What are the naming conventions? 
If you cannot defined them then remove this

File: frontend/ui-guidelines.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+## Accessibility
+- We prioritize accessibility, ensuring that all components are usable by everyone.
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:24]: refer to ## Accessibility Testing of the read me

File: frontend/ui-guidelines.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (ApostolosBenisis) [inline frontend/ui-guidelines.md:25]: Additionally, we have a Styles Story in Storybook to visualize our palette and provide visual guidance for selecting the appropriate style elements. For further guidance, refer to the Material Design System v2.

**Review by ApostolosBenisis — CHANGES_REQUESTED** [review]

**Pull request merged**